pkg/cloud/services/ec2/gateways.go

```diff
@@ -127,18 +131,6 @@ func (s *Service) createInternetGateway() (*ec2.InternetGateway, error) {
 	s.scope.Info("Created internet gateway for VPC", "vpc-id", s.scope.VPC().ID)
 
 	tagParams := s.getGatewayTagParams(*ig.InternetGateway.InternetGatewayId)
-	if err := wait.WaitForWithRetryable(wait.NewBackoff(), func() (bool, error) {
-		if err := tags.Apply(&tags.ApplyParams{
-			EC2Client:   s.scope.EC2,
-			BuildParams: tagParams,
-		}); err != nil {
-			return false, err
-		}
-		return true, nil
-	}, awserrors.InternetGatewayNotFound); err != nil {
-		record.Warnf(s.scope.AWSCluster, "FailedTagInternetGateway", "Failed to tag managed Internet Gateway %q: %v", *ig.InternetGateway.InternetGatewayId, err)
-		return nil, errors.Wrapf(err, "failed to tag internet gateway %q", *ig.InternetGateway.InternetGatewayId)
-	}
 
 	// Update the tags, so that when ig.InternetGateway is returned it has the
 	// latest tag data rather than returning empty tags.
```

Comment:
Is the below update to `ig.InternetGateway.Tags` still needed? I would expect the returned internet gateway to contain the tags now that it's tagging on create.

Reply:
For followup, but we should probably create an issue to make attaching the gateway re-entrant, looks like we only attempt to attach when creating.